app/api/dds/v1/auth_providers_api.rb

```diff
@@ -28,6 +57,113 @@ class AuthProvidersAPI < Grape::API
       get '/auth_providers/:id', root: false, serializer: AuthenticationServiceSerializer do
         AuthenticationService.find(params[:id])
       end
+
+      desc 'List Auth Provider Affiliates' do
+        detail 'List Auth Provider Affiliates'
+        named 'list authentication provider affiliates'
+        failure [
+          [200, 'Success'],
+          [404, 'AuthenticationService does not exist'],
+          [400, 'AuthenticationService does not support affiliate search'],
+          [503, 'Identity Provider Communication Failure']
+        ]
+      end
+      params do
+        requires :id, type: String, desc: 'AuthenticationProvider UUID'
+        requires :full_name_contains, type: String, desc: 'string contained in name(must be at least 3 characters)'
+      end
+      get '/auth_providers/:id/affiliates', root: 'results', each_serializer: AffiliateSerializer do
+        affiliate_params = declared(params, {include_missing: false}, [:full_name_contains])
+        if UserAuthenticationService.where(
+          authentication_service_id: params[:id],
+          uid: affiliate_params[:uid]
+        ).exists?
+          affiliate_exists_error!
+        else
+          auth_service = AuthenticationService.find(params[:id])
+          if auth_service.identity_provider.nil?
+            unsupported_affiliate_search_error!
+          else
+            auth_service.identity_provider.affiliates(affiliate_params[:full_name_contains])
```

Comment:
Hmm, I think it might be best to add pagination in now, if possible.  If spike of LDAP pagination is going to take weeks vs days, then i say we can skip it.  But if we can incorporate it now, I think thats better than not.  We can weigh the cost of development now vs cost later.